infra/util/test_helper/MockCallback.hpp

```diff
@@ -124,6 +124,34 @@ namespace infra
         }
     };
 
+    template<class Result>
+    class NotifyingVerifyingFunctionMock;
+
+    template<class Result, class... Args>
+    class NotifyingVerifyingFunctionMock<Result(Args...)>
+    {
+    public:
+        explicit NotifyingVerifyingFunctionMock(infra::Function<Result(Args...)> onCalled)
+            : onCalled(onCalled)
+        {
+            EXPECT_CALL(*this, callback());
+        }
+
+        MOCK_CONST_METHOD0_T(callback, Result());
```

Comment:
You could use the newer macros for mocking methods:
```suggestion
        MOCK_METHOD(Result, callback, (), (const override));
```